### PR TITLE
tests: Remove unused includes

### PR DESCRIPTION
**Pull request by practicalswift**
Reduce compilation time and unneccessary recompiles by removing unused includes in tests.

A subset of #16273 ("refactor: Reduce total compilation time by 2% and avoid unnecessary recompiles by removing unused includes") as requested by MarcoFalke in https://github.com/bitcoin/bitcoin/pull/16273#issuecomment-505022643.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
ACK 30dfe9a86c29e6e95aac236b9273738e392cf86e (checked that added includes are required and that removed includes are not)

**Comment by fanquake** [comment]
47a15fa47519a5ab16a3d6c1dd3227dd76076365 does not compile.

```bash
 CCLD     libsecp256k1.la
  CXX      test/test_bitcoin-mempool_tests.o
test/fs_tests.cpp:14:26: error: use of undeclared identifier 'GetDataDir'
    fs::path tmpfolder = GetDataDir();
                         ^
1 error generated.
make[2]: *** [test/test_bitcoin-fs_tests.o] Error 1
make[2]: *** Waiting for unfinished jobs....
test/flatfile_tests.cpp:16:27: error: use of undeclared identifier 'GetDataDir'
    const auto data_dir = GetDataDir();
                          ^
test/flatfile_tests.cpp:29:27: error: use of undeclared identifier 'GetDataDir'
    const auto data_dir = GetDataDir();
                          ^
test/flatfile_tests.cpp:90:27: error: use of undeclared identifier 'GetDataDir'
    const auto data_dir = GetDataDir();
                          ^
test/flatfile_tests.cpp:110:27: error: use of undeclared identifier 'GetDataDir'
    const auto data_dir = GetDataDir();
                          ^
4 errors generated.
make[2]: *** [test/test_bitcoin-flatfile_tests.o] Error 1
make[1]: *** [check-recursive] Error 1
make: *** [check-recursive] Error 1
```

**Comment by practicalswift** [comment]
@fanquake Yes I know, I'm fixing that as we speak! :-)

File: src/qt/test/test_main.cpp [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include <qt/test/uritests.h>
 #include <qt/test/compattests.h>
 #include <test/setup_common.h>
+#include <util/time.h>
```
Comment (MarcoFalke) [inline src/qt/test/test_main.cpp]: No longer needed after rebase

Comment (practicalswift) [reply src/qt/test/test_main.cpp]: Good catch!

Re-ran my analysis script and it turns out that `util/time.h` and `chainparams.h` are no longer needed.

**Comment by Sjors** [comment]
ACK 9a84169 on macOS 10.14.5 (I rebased on #16289)

**Pull request merged**